### PR TITLE
feat: llm error messages in preview endpoint (PL-931)

### PR DESCRIPTION
**Pull request by theProf**
### Brief description. What is this change?

Return LLM error messages in preview endpoint for display in toasts

### CODE REVIEW

File: lib/controllers/test/interface.ts [hunk 1]
```diff
@@ -75,6 +75,13 @@ export const TestClassificationResponseDTO = z
     nlu: ClassificationDTO,
     llm: ClassificationDTO,
     utterance: z.string(),
+    errors: z.optional(
+      z.array(
```
Comment (pmvrmc) [inline lib/controllers/test/interface.ts:79]: you can do `z.array().optional()` and avoid the extra wrapper

File: lib/controllers/test/index.ts [hunk 2]
```diff
@@ -212,6 +212,9 @@ class TestController extends AbstractController {
       ];
     })();
 
+    // Only returning LLM related error message for now (PL-931)
+    const errors = predictions.llm?.error && [predictions.llm.error];
```
Comment (pmvrmc) [inline lib/controllers/test/index.ts:216]: nit: mentally the ternary operator is easier for me, somethinglike 
`const errors = predictions.llm?.error ? [predictions.llm.error] : undefined;`

but your solution works fine!

**Review by pmvrmc — APPROVED** [review]

File: lib/controllers/test/index.ts (hunk 2 above)
Comment (pmvrmc) [reply lib/controllers/test/index.ts:216]: you could even inline this since its not used anywhere else

Comment (theProf) [reply lib/controllers/test/index.ts:216]: i think i'll leave this one, but that's good to know!

**Pull request merged**